src/view/renderer.js

```diff
@@ -639,6 +652,42 @@ export default class Renderer {
 			}
 		}
 	}
+
+	/**
+	 * Checks if two given selections are similar. Selections are considered similar if they are non-collapsed
+	 * and their trimmed (see {@link #_trimSelection}) representations are equal.
+	 *
+	 * @private
+	 * @param {module:engine/view/selection~Selection} current
+	 * @param {module:engine/view/selection~Selection} previous
+	 * @returns {Boolean}
+	 */
+	_areSimilarSelections( current, previous ) {
+		return !current.isCollapsed && this._trimSelection( current ).isEqual( this._trimSelection( previous ) );
+	}
+
+	/**
+	 * Creates a copy of a given selection with all of its ranges
+	 * trimmed (see {@link module:engine/view/range~Range#getTrimmed getTrimmed}).
+	 *
+	 * @private
+	 * @param {module:engine/view/selection~Selection} selection
+	 * @returns {module:engine/view/selection~Selection} Selection copy with all ranges trimmed.
+	 */
+	_trimSelection( selection ) {
+		const newSelection = Selection.createFromSelection( selection );
```

Comment:
`createFromSelection()` gives us `setTo()` call which e.g. copies fake selection's stuff so that's good. What I like less is the fact that we are copying the ranges here (`getRanges()` copies them) to copy them again when doing `setRanges()` to copy them again 3 times on another `getRanges()`, `getTrimmes()`, `setRanges()`. This means 6 times for every selection and we need to trim two selections...